### DISCUSSION TITLE
Activate python virtual environment in pane

### DISCUSSION THREAD

**Discussion by ParkerKain** [zellij-org/zellij#3310] [Q&A]
Hey! Loving `zellij` so far! 

For python development, I very regularly want to create a new pane, and then source some virtual environment via `source .venv/bin/activate`, which attaches me to a virtual environment for running other commands via the shell. I want to do this for all of my `zellij` panes, for running tests, opening my editor, etc. 

However, it looks like if I configure a layout via `command` to run the `source` command, I don't get spit back out into my shell, I am in the "command mode" - and if I exit it, I am not attached to my virtual environment. 

Is there a standard way of doing this I am missing? Thanks!

**Comment by susludusrotalis** [comment]
Run into the same problem. I suppose that is because source is a Bash builtin and as Zellij doc states:
> [command is a string (path) to an executable that should be run in this pane instead of the default shell.](https://zellij.dev/documentation/creating-a-layout)

I've also tried the following:
`bash --rcfile <(cat ~/.bashrc; echo 'source venv/bin/activate';)`
Here Bash <(command_list) syntax (called [Process Substitution](https://tldp.org/LDP/abs/html/process-sub.html)) creates temp file and redirects process output to it.
That works fine in bash shell, but not when I`m trying to use it in predifined Zellij layout:
```
layout {
    pane command="bash" {
        args "--rcfile" "<(cat ~/.bashrc; echo 'source venv/bin/activate';)"
    }
}
```

**Reply by susludusrotalis** [reply]
@ParkerKain Found a way to source virtualenv like this:
```
layout {
    pane command="bash" {
        args "-c" "bash --rcfile <(cat ~/.bashrc; echo 'source venv/bin/activate';)"
    }
}
```
https://github.com/zellij-org/zellij/issues/3361#issuecomment-2158502220
